datafusion/core/src/sql/planner.rs

```diff
@@ -524,7 +524,18 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     keys.into_iter().unzip();
 
                 // return the logical plan representing the join
-                if filter.is_empty() {
+                if left_keys.is_empty() {
+                    // When we don't have join keys, use cross join
+                    let join = LogicalPlanBuilder::from(left).cross_join(&right)?;
+
+                    let join_filter_init = filter.remove(0);
+                    join.filter(
+                        filter
+                            .into_iter()
+                            .fold(join_filter_init, |acc, e| acc.and(e)),
+                    )?
+                    .build()
```

Comment:
I wonder if you can use `reduce` here rather than fold: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.reduce

Something like (untested):

```suggestion
                    join.filter(
                        filter
                            .reduce(|acc, e| acc.and(e)),
                    )?
                    .build()
```